tls/s2n_record_write.c

```diff
@@ -64,25 +64,15 @@ static int s2n_tls_record_overhead(struct s2n_connection *conn)
     return extra;
 }
 
-/* Limits the size of the fragment length to the maximum allowed by the protocol */
-static int s2n_record_limit_write_payload_size(struct s2n_connection *conn, uint16_t size_without_overhead)
-{
-    int max_fragment_size = MIN(size_without_overhead, S2N_TLS_MAXIMUM_FRAGMENT_LENGTH);
-
-    return max_fragment_size;
-}
-
 /* This function returns maximum size of plaintext data to write for the payload.
- * Record overheads are not calculated here but they have been accounted as part of the total record sizes.
+ * Record overheads are not included here.
  */
-int s2n_record_max_write_payload_size(struct s2n_connection *conn)
+uint16_t s2n_record_max_write_payload_size(struct s2n_connection *conn)
 {
-    int bytes;
-    GUARD(bytes = s2n_record_limit_write_payload_size(conn, conn->max_outgoing_fragment_length));
+    /* Limits the fragment length to the maximum allowed by TLS */
+    const uint16_t max_fragment_size = MIN(conn->max_outgoing_fragment_length, S2N_TLS_MAXIMUM_FRAGMENT_LENGTH);
```

Comment:
This should probably check for a null `conn`

```suggestion
    const uint16_t max_outgoing_fragment_length = conn ? conn->max_outgoing_fragment_length : 0;
    const uint16_t max_fragment_size = MIN(max_outgoing_fragment_length, S2N_TLS_MAXIMUM_FRAGMENT_LENGTH);
```

Reply:
hmm I'm now thinking reverting to the previous s2n_record_max_write_payload_size() interface.. returning 0 would cause downstream issues if not handled as an error

Reply:
Ah right, it's using `MIN`. Could we set the default to `S2N_TLS_MAXIMUM_FRAGMENT_LENGTH` instead of `0`?